Consul.Test/ConnectTest.cs

```diff
@@ -76,5 +79,94 @@ public async Task Connect_CASetConfig()
+                        Name = "Medhi",
+                        Action = "allow",
+                        LegacyCreateTime = DateTime.UtcNow,
+                        LegacyUpdateTime = DateTime.UtcNow,
+                    }
+                }
+            };
+            var secondEntry = new ServiceIntentionsEntry
+            {
+                Kind = "service-intentions",
+                Name = "Second",
+                Sources = new List<SourceIntention>
+                {
+                    new SourceIntention
+                    {
+                        Name = "Optimus-Prime",
+                        Action = "allow",
+                        LegacyCreateTime = DateTime.UtcNow,
+                        LegacyUpdateTime = DateTime.UtcNow,
+                    },
+                    new SourceIntention
+                    {
+                        Name = "Megatron",
+                        Action = "deny",
+                        LegacyCreateTime = DateTime.UtcNow,
+                        LegacyUpdateTime = DateTime.UtcNow,
+                    },
+                    new SourceIntention
+                    {
+                        Name = "Sentinel-Prime",
+                        Action = "deny",
+                        LegacyCreateTime = DateTime.UtcNow,
+                        LegacyUpdateTime = DateTime.UtcNow,
+                    }
+                }
+            };
+            var resultOne = await _client.Configuration.ApplyConfig(firstEntry);
+            Assert.Equal(HttpStatusCode.OK, resultOne.StatusCode);
+
+            var resultTwo = await _client.Configuration.ApplyConfig(secondEntry);
+            Assert.Equal(HttpStatusCode.OK, resultTwo.StatusCode);
+
+            var intentionsQuery = await _client.Connect.ListIntentions<ServiceIntention>();
+            Assert.Equal(HttpStatusCode.OK, intentionsQuery.StatusCode);
+
+            var intentions = intentionsQuery.Response;
+            Assert.NotNull(intentions);
+
+            Assert.Contains(intentions, i => !string.IsNullOrEmpty(i.DestinationName));
+            Assert.Contains(intentions, i => !string.IsNullOrEmpty(i.SourceName));
+            Assert.Contains(intentions, i => !string.IsNullOrEmpty(i.DestinationNS));
+            Assert.Contains(intentions, i => !string.IsNullOrEmpty(i.SourceType));
+            Assert.Contains(intentions, i => !string.IsNullOrEmpty(i.SourceNS));
+            Assert.Contains(intentions, i => !string.IsNullOrEmpty(i.Action) && i.Action == "allow" || i.Action == "deny");
+            Assert.Contains(intentions, i => i.CreateIndex > 0);
+            Assert.Contains(intentions, i => i.ModifyIndex > 0);
+            Assert.Contains(intentions, i => i.Precedence > 0);
+            Assert.Contains(intentions, i => i.SourceName == "fortunate" && i.DestinationName == firstEntry.Name);
+            Assert.Contains(intentions, i => i.SourceName == "Optimus-Prime" && i.DestinationName == secondEntry.Name);
```

Comment:
Generally, these assertions are OK, but they are slightly more complicated than necessary.
It would be better to assign one of the intentions to a variable and then 
write assertions like this:
```
var int1 = intentions.First(i => i.SourceName == "Optimus-Prime");
Assert.NotEmpty(int1.DestinationName);
Assert.True(int1.CreateIndex > 0);
...
```